apps/docs/pages/docs/api-reference/fields/richtext.mdx

```diff
@@ -0,0 +1,733 @@
+        },
+      },
+    },
+  },
+};
+```
+
+### `controls`
+
+Controls let you swap or extend the buttons available to each menu key. A control is an object with a `render(editor, editorState)` function, so you can plug in any React UI while still receiving the merged TipTap state produced by our selector utilities.
+
+```tsx {1-12, 20-22} copy
+const BoldControl = {
+  render: (editor, editorState) => (
+    <button
+      type="button"
+      onClick={(e) => {
+        e.stopPropagation();
+        editor.chain().focus().toggleBold().run();
+      }}
+      disabled={!editorState.canBold}
+      style={{ fontWeight: editorState.isBold ? "bold" : "normal" }}
+    >
+      Bold
+    </button>
+  ),
+};
+
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          controls: {
+            Bold: BoldControl,
+          },
+        },
+      },
+    },
+  },
+};
+```
+
+<Callout>
+  Note: The `e.stopPropagation();` is required in Puck in order to avoid issues
+  with the portal on inlineMenu.
+</Callout>
+
+### `textSelectOptions`
+
+Configure the block-style dropdown rendered in the toolbar. Supply any subset of `"p" | "h1" … "h6"` to narrow or expand the list.
+
+```tsx {7} copy
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          textSelectOptions: ["p", "h2", "h3"],
+        },
+      },
+    },
+  },
+};
+```
+
+### `extensions`
+
+Our RichText editor is powered by [TipTap](https://tiptap.dev/). The `extensions` option accepts any TipTap extension instances, allowing you to enable new nodes or marks alongside the default extensions.
+
+```tsx {1, 9} copy
+import Link from "@tiptap/extension-link";
+
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          extensions: [Link],
+        },
+      },
+    },
+  },
+};
+```
+
+<Callout type="info">
+  TipTap maintains extensive [extension
+  guides](https://tiptap.dev/docs/editor/extensions/overview). Any configuration
+  recommended there works inside your `extensions` array and via `controls`,
+  `selector` + `menu`/`inlineMenu` overrides.
+</Callout>
+
+### `selector`
+
+Pass a custom selector `(ctx) => ({ ... })` to add booleans into the merged editor state. This is useful when supplying bespoke controls that need to react to TipTap attributes or mark status beyond what the defaults provide.
+
+```tsx {1-8, 16} copy
+const highlightSelector = (ctx) => {
+  const editor = ctx.editor;
+  if (!editor) return {};
+  return {
+    isHighlight: editor.isActive("highlight"),
+    canHighlight: editor.can().chain().toggleHighlight().run(),
+  };
+};
+
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          selector: highlightSelector,
+        },
+      },
+    },
+  },
+};
+```
+
+## Default values
```

Comment:
I think this section is generally unnecessary, and most defaults can be folded into the parameters. We don't do this elsewhere and the level of detail is a bit on the verbose side.